### ISSUE TITLE
Emmet is not working at all. Sublime text 3126, I am using Ubuntu 16.04 64-bit

### ISSUE THREAD

**Issue by atatof** [sergeche/emmet-sublime#637]
Hey guys, emmet is not working at all. I installed emmet and Pyv8 manually  and nothing happened.
I downloaded all the file from this link https://github.com/sergeche/emmet-sublime#readme
Please help!

**Comment by AndersSchmidtHansen** [comment]
I think I have the same issue (on ElementaryOS 14.04), though I did not install the Pyv8 manually. I re-installed everything through PackageControl. The log that I get is this: 

```
Traceback (most recent call last):
  File "/opt/sublime_text/sublime_plugin.py", line 109, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/sublime_plugin.py", line 915, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "emmet-plugin in /home/user/.config/sublime-text-3/Installed Packages/Emmet.sublime-package", line 24, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/home/user/.config/sublime-text-3/Installed Packages/Emmet.sublime-package/emmet/pyv8loader.py", line 20, in <module>
AttributeError: 'module' object has no attribute 'error'
```

Maybe that can shed some light on what could be the issue. :+1:

**Comment by Kout** [comment]
The workaround suggested on [Sublime Forum](https://forum.sublimetext.com/t/solved-emmet-not-working-in-3124/23122) works, but needs to be repeated after every restart and Sublime opens with following warning (at least in my case ‒ might depend on installed plugins): 
```
2 missing dependencies were just installed. Sublime Text should be restarted, otherwise one or more of the installed packages may not function properly.
Exception in thread Thread-25:
Traceback (most recent call last):
  File "./python3.3/threading.py", line 901, in _bootstrap_inner
  File "./python3.3/threading.py", line 1142, in run
  File "/home/kout/.config/sublime-text-3/Packages/JSHint Gutter/JSHint.py", line 188, in <lambda>
    self.timer = Timer(timeout, lambda: view.window().run_command("jshint", { "show_panel": False }))
AttributeError: 'NoneType' object has no attribute 'run_command'
```

**Comment by cgslivre** [comment]
Complete log from my Sublime. I'm running on Ubuntu 14.04 LTS

DPI scale: 1
startup, version: 3126 linux x64 channel: stable
executable: /opt/sublime_text/sublime_text
working dir: /
packages path: /home/marcio/.config/sublime-text-3/Packages
state path: /home/marcio/.config/sublime-text-3/Local
zip path: /opt/sublime_text/Packages
zip path: /home/marcio/.config/sublime-text-3/Installed Packages
ignored_packages: ["AutoAligner", "Autoprefixer", "Bootstrap 3 Snippets", "BracketHighlighter", "bs3-sublime", "CSS Extended Completions", "Function Name Display", "Git", "Material Theme - Appbar", "Material Theme - White Panels", "PlainTasks", "Preferences Editor", "Project PHP ClassBrowser", "SideBarFolders", "SidebarSeparator", "Themes Menu Switcher", "Vintage"]
pre session restore time: 0.333253
warning: mnemonic t not found in menu caption Tools
startup time: 0.478787
first paint time: 0.529367
first paint time: 0.781414
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.01-ssl-linux
Package Control: Linux SSL: successfully loaded _ssl module for libssl.so.1.0.0
Package Control: Linux SSL: unloading http module so ssl will be available
Package Control: Linux SSL: unloading urllib module so ssl will be available
reloading plugin 0_package_control_loader.50-backrefs
reloading plugin 0_package_control_loader.50-markupsafe
reloading plugin 0_package_control_loader.50-python-markdown
reloading plugin 0_package_control_loader.51-python-jinja2
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin AdvancedNewFile.AdvancedNewFile
reloading plugin Alignment.Alignment
reloading plugin All Autocomplete.all_views_completions
reloading plugin AutoFileName.autofilename
reloading plugin AutoFileName.getimageinfo
reloading plugin BufferScroll.BufferScroll
reloading plugin CSS3.css3_completions
reloading plugin CSS3.rgb_to_hex
reloading plugin DocBlockr.jsdocs
reloading plugin Emmet.emmet-plugin
Traceback (most recent call last):
  File "/opt/sublime_text/sublime_plugin.py", line 109, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/opt/sublime_text/sublime_plugin.py", line 915, in load_module
    exec(compile(source, source_path, 'exec'), mod.__dict__)
  File "emmet-plugin in /home/marcio/.config/sublime-text-3/Installed Packages/Emmet.sublime-package", line 24, in <module>
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "/home/marcio/.config/sublime-text-3/Installed Packages/Emmet.sublime-package/emmet/pyv8loader.py", line 20, in <module>
AttributeError: 'module' object has no attribute 'error'
reloading plugin FileDiffs.file_diffs
reloading plugin HTML5.encode_html_entities
reloading plugin HTML5.html_completions
reloading plugin JavaScriptFunctionDefinition.JavaScriptFunctionDefinition
reloading plugin Keymaps.Keymaps
reloading plugin List stylesheet variables.stylevariables
reloading plugin LoremGibson.Loremgibson
reloading plugin Material Theme.mt_activator
reloading plugin Material Theme.mt_changes
reloading plugin Material Theme.mt_config
reloading plugin Material Theme.mt_info
reloading plugin Material Theme.mt_urls
reloading plugin Nodejs.Nodejs
reloading plugin Nodejs.__init__
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin ProjectManager.closewindow
reloading plugin ProjectManager.pm
reloading plugin SideBarEnhancements.SideBar
reloading plugin SideBarEnhancements.SideBarAPI
reloading plugin SideBarEnhancements.SideBarDefaultDisable
reloading plugin StatusMessage.commands
reloading plugin StatusMessage.default_status
reloading plugin StatusMessage.status_message
reloading plugin SublimeLinter-php.linter
SublimeLinter: php linter loaded 
reloading plugin Sync View Scroll.syncscroll
reloading plugin ColorPicker.sublimecp
reloading plugin EncodingHelper.EncodingHelper
reloading plugin Gulp.base_command
reloading plugin Gulp.caches
reloading plugin Gulp.cross_platform_codecs
reloading plugin Gulp.cross_platform_process
reloading plugin Gulp.dir_context
reloading plugin Gulp.event_task
reloading plugin Gulp.gulp
reloading plugin Gulp.gulp_version
reloading plugin Gulp.hasher
reloading plugin Gulp.insert_in_output_view
reloading plugin Gulp.plugins
reloading plugin Gulp.progress_notifier
reloading plugin Gulp.settings
reloading plugin Gulp.timeout
reloading plugin HTML-CSS-JS Prettify.HTMLPrettify
reloading plugin Minify.Minify
reloading plugin NodeRequirer.NodeRequirer
reloading plugin SFTP.SFTP
reloading plugin StringUtilities.stringutilities
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.sublimelinter
reloading plugin copy-file-name.copy_filename
plugins loaded
SublimeLinter: debug mode: off 
SublimeLinter: php activated: /usr/bin/php 
syncScroll starting
warning: mnemonic t not found in menu caption Tools
Package Control: Skipping automatic upgrade, last run at 2016-09-26 12:11:08, next run at 2016-09-26 13:11:08 or after

**Comment by rafapaulin** [comment]
same here

**Comment by atatof** [comment]
The only way that I could fix it was installing Ubuntu again!

**Comment by stilliard** [comment]
same issue here

**Comment by SubodhDahal** [comment]
It seems that the `urllib` Python module that is shipped with recent builds of Sublime Text  3 doesn't have the `url_err` and `url_parse` methods. It only contains the `request` method.

For now I've implemented a hacky solution in my system, which will work on the cases that there's no errors while downloading the pyv8 library. I cloned the repository to the  sublime packages folder and made changes in the `emmet-sublime/emmet/pyv8loader.py` file at line 20 and 21 to:

```
url_err = ''
url_parse = ''
```

It works at multiple restarts as well. But we'll have to find a better solution for this problem.
